satpy/readers/hdf5_utils.py

```diff
@@ -120,3 +122,14 @@ def get(self, item, default=None):
             return self[item]
         else:
             return default
+
+
+def from_h5_array(h5dset):
+    """Create a dask array from an h5py dataset, ensuring uniqueness of the dask array name."""
+    chunk_size = dc.get("array.chunk-size")
+
+    chunks = normalize_chunks(chunk_size, dtype=h5dset.dtype, previous_chunks=h5dset.chunks, shape=h5dset.shape)
+    name = tokenize(os.fspath(h5dset.file.filename), h5dset.name, chunks)
```

Comment:
My only last optional request is that `h5dset.name` be added as a prefix to the `name`. So `h5dset.name + tokenize(...)` so that when debugging (dask diagnostics, task graphs, etc) we would know which variable is being used instead of a relatively arbitrary token/hash.

Reply:
Good idea, fixed.